packages/edit-site/src/store/actions.js

```diff
@@ -257,3 +264,84 @@ export function updateSettings( settings ) {
 		settings,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	if ( ! isTemplateRevertable( template ) ) {
+		yield controls.dispatch(
+			'core/notices',
+			'createErrorNotice',
+			__( 'This template is not revertable.' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		yield controls.dispatch(
+			'core',
+			'deleteEntityRecord',
```

Comment:
We're not removing the record, we're modifying it. (the record for the client is not the CPT post) It's weird to have to call "deleteEntityRecord" to perform this, especially since the template actually stays there and removing from the state doesn't make sense, for me it should be more a "put" request. So I wonder if we  need a  dedicated endpoint instead or if we could reuse the  "updateEntityRecord" instead.

Reply:
I think `deleteEntityRecord` is necessary if we want to make sure the file-based template is loaded. If we just update the CPT with the original file's content, then the CPT will still stay there and it will be used rather than the file-based template.

This means, after updating the theme - assuming the file template changes - the CPT will be loaded rather than the updated file template.

Reply:
I think you misunderstood my comment. I think we should remove the CPT entry indeed but we shouldn't be calling "deleteEntityRecord" to do so because we're actually updating the record (from the client's perspective). Basically what we're doing here is changing the "isCustom" property so I wonder if the right approach is to:

```
editEntityRecord( { 'isCustom': isCustom } )
saveEntityRecord()
``` 

and have the endpoint to edit template work properly when you "set isCustom". Meaning if you set isCustom to false, it should actually remove the CPT entry and return the file on in response.

Reply:
Gotcha! Yep, I totally misunderstood you! Thanks for the clarification.

It sounds great to me! It would also get rid of the "empty editor" flash caused by the `deleteEntityRecord` + refetch.